src/main.rs

```diff
@@ -11,25 +7,36 @@ use std::io::Write;
 use std::process;
 
 use gb_io::reader::SeqReader;
-use regex::Regex;
-
-// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
-// Indexing into this array is as follows:
-// T = 0, C = 1, A = 2, G = 3
-// index = (16 * first_base) + (4 * second_base) + third_base
-// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
-const GENETIC_CODE: &str = "FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
-// some codons can be used as Met in the start position
-const STARTS: &str = "---M------**--*----M------------MMMM---------------M------------";
-
-// ERR_BAD_NT is an error value for an invalid nucleotide
+use gb_io::{feature_kind, qualifier_key};
+use once_cell::sync::Lazy;
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Something went wrong in translation! {0:?}")]
+    BadNucleotide(Vec<usize>),
+
+    #[error("Invalid amino acid: {0}")]
+    InvalidAminoAcid(char),
+}
+
+/// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
+/// Indexing into this array is as follows:
+/// T = 0, C = 1, A = 2, G = 3
+/// index = (16 * first_base) + (4 * second_base) + third_base
+/// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
+const GENETIC_CODE: &[u8] = b"FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
```

Comment:
I replaced the `&str` usage with `&[u8]` because it is easier to deal for the use case of this program. `&str` and `String` in Rust use the same "storage" as `&[u8]` and `Vec<u8>` (respectively), but they are stricter because each element must be valid UTF-8. Because for this application you're only working with a reduced alphabet (that is also representable in ASCII), it is easier to work with the bytes directly.